onnxruntime/core/providers/cuda/tensor/nonzero_impl.cu

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "nonzero_impl.h"
+#include "core/providers/cuda/shared_inc/cuda_call.h"
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include <cub/cub.cuh>
+
+namespace onnxruntime {
+namespace cuda {
+
+static const int THREADS_PER_BLOCK = GridDim::maxThreadsPerBlock;
+
+int NonZeroCalcBlockCount(int64_t x_size) 
+{
+  return CeilDiv(x_size, THREADS_PER_BLOCK);
+}
+
+size_t NonZeroCalcPrefixSumTempStorageBytes(int* prefix_counts, int number_of_blocks)
+{
+  size_t temp_storage_bytes = 0;
+  CUDA_CALL_THROW(cub::DeviceScan::InclusiveSum(
+      nullptr, temp_storage_bytes, prefix_counts, prefix_counts, number_of_blocks));
+  return temp_storage_bytes;
+}
+
+void NonZeroInclusivePrefixSum(void* d_temp_storage, size_t temp_storage_bytes, int* prefix_counts, int number_of_blocks)
+{
+  CUDA_CALL_THROW(cub::DeviceScan::InclusiveSum(
```

Comment:
>CUDA_CALL_THROW [](start = 2, length = 15)

I'd suggest not using throw as it causes crash without proper logging. Can you use CUDA_CALL and check its bool return value for early termination on error?

Reply:
Will try a better one on this, maybe put the error_code return to operator compute() part to do the check.